python/az/aro/azext_aro/custom.py

```diff
@@ -150,7 +182,37 @@ def aro_list_credentials(client, resource_group_name, resource_name):
     return client.list_credentials(resource_group_name, resource_name)
 
 
-def aro_update(client, resource_group_name, resource_name, no_wait=False):
+def aro_update(cmd, client, resource_group_name, resource_name, no_wait=False):
+    oc = client.get(resource_group_name, resource_name)
+
+    master_subnet = oc.master_profile.subnet_id
+    worker_subnets = {w.subnet_id for w in oc.worker_profiles}
+
+    master_parts = parse_resource_id(master_subnet)
+    vnet = resource_id(
+        subscription=master_parts['subscription'],
+        resource_group=master_parts['resource_group'],
+        namespace='Microsoft.Network',
+        type='virtualNetworks',
+        name=master_parts['name'],
+    )
+
+    aad = AADManager(cmd.cli_ctx)
+    rp_client_id = os.environ.get('AZURE_FP_CLIENT_ID', FP_CLIENT_ID)
+    rp_client_sp = aad.get_service_principal(rp_client_id)
+
+    c_client_id = oc.service_principal_profile.client_id
+    c_client_sp = aad.get_service_principal(c_client_id)
```

Comment:
The client SPN may end up deleted by the customer - we need to check for this case.

Reply:
Implemented a check which displays a warning as we (currently) don't handle cluster service principal deletion.